app/controllers/application_controller.rb

```diff
@@ -74,4 +74,8 @@ def user_not_authorized
       redirect_back(fallback_location: root_path)
     end
   end
+
+  def unwrap_pundit_scope(scope)
+    scope.is_a?(Array) ? scope : [scope]
+  end
```

Comment:
When assigning namespaces to pundit via records and scopes, we are expected to either pass an array of symbols and then the record or scope:

```ruby
policy([namespace, record])
```

The Pundit policy finder uses the last element in the case an array is given as the record: https://github.com/varvet/pundit/blob/main/lib/pundit/context.rb#L173-L175

In the case above, and following Pundit's guidance, this is true. https://github.com/varvet/pundit?tab=readme-ov-file#policy-namespacing

The issue we have occurs with multiple layers of recursion:

```ruby
class Claims::ApplicationController
  def policy(record)
    super([:claims, record])
  end
end
```

This calls the top-level `pundit.policy` with `[:claims, record]`.

But then we do:

```ruby
class Claims::Support::ApplicationController < Claims::ApplicationController
  def policy(record)
    super([:support, record])
  end
end
```

This is to insert the next namespace level into the namespaces, but what we end up calling would be `pundit.policy` with `[:claims, [:support, record]]`. This makes the last element of the array an array itself.

We want to flatten this hierarchy of namespaces. To do this, the `unwrap_pundit_scope` method is introduced.

This might be better named `wrap_pundit_scope` actually...?

Then we flatten the given array or wrapped non-array resource to pass it through.

e.g.

`Claims::Support::ApplicationController#policy` unfolds it's defined `policy` method as follows:


```ruby
policy([:something, claim]) # `claim` is of type `Claims::Claim`
=> policy([:claims, :support, :something, claim]) # Typically `Claims::Claim` would become `Claims::ClaimPolicy`, but we already define the service namespace at the controller, we don't need to use the `Claims` namespace from the model.
=> Claims::Support::Something::ClaimPolicy
```